From 61535b58452a18d6654a02fd25bfbc562bbbdbea Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Wed, 22 Feb 2017 17:23:42 -0800
Subject: [PATCH] Add an option to disable the tk module

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Signed-off-by: Samuel Martin <s.martin49@gmail.com>
[ Andrey Smirnov: ported to Python 3.6 ]
Signed-off-by: Andrey Smirnov <andrew.smirnov@gmail.com>
[ Adam Duskett: ported to Python 3.10.0 ]
Signed-off-by: Adam Duskett <aduskett@gmail.com>
[ Bernd Kuhls: ported to Python 3.11.4]
Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>
---
 Makefile.pre.in | 11 +++++++----
 configure.ac    |  9 +++++++++
 2 files changed, 16 insertions(+), 4 deletions(-)

diff --git a/Makefile.pre.in b/Makefile.pre.in
index 1c7b38ce4f6..6ab184fe5f8 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -1933,7 +1933,6 @@ LIBSUBDIRS=	asyncio \
 		multiprocessing multiprocessing/dummy \
 		re \
 		site-packages \
-		tkinter \
 		tomllib \
 		turtledemo \
 		unittest \
@@ -2060,9 +2059,6 @@ TESTSUBDIRS=	ctypes/test \
 		test/xmltestdata \
 		test/xmltestdata/c14n-20 \
 		test/ziptestdata \
-		tkinter/test \
-		tkinter/test/test_tkinter \
-		tkinter/test/test_ttk \
 		unittest/test \
 		unittest/test/testmock
 
@@ -2082,6 +2078,13 @@ ifeq (@SQLITE3@,yes)
 LIBSUBDIRS += sqlite3
 endif
 
+ifeq (@TK@,yes)
+LIBSUBDIRS += tkinter
+TESTSUBDIRS += tkinter/test tkinter/test/test_tkinter \
+	tkinter/test/test_ttk
+endif
+
+
 TEST_MODULES=@TEST_MODULES@
 libinstall:	all $(srcdir)/Modules/xxmodule.c
 	@for i in $(SCRIPTDIR) $(LIBDEST); \
diff --git a/configure.ac b/configure.ac
index 0013afc9aad..342b295bcdd 100644
--- a/configure.ac
+++ b/configure.ac
@@ -4231,6 +4231,15 @@ if test "$SQLITE3" = "no" ; then
    DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} _sqlite3"
 fi
 
+AC_SUBST(TK)
+AC_ARG_ENABLE(tk,
+	AS_HELP_STRING([--disable-tk], [disable tk]),
+	[ TK="${enableval}" ], [ TK=yes ])
+
+if test "$TK" = "no"; then
+   DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} _tkinter"
+fi
+
 AC_SUBST(PYDOC)
 
 AC_ARG_ENABLE(pydoc,
-- 
2.44.0

